spec/lib/sprig_spec.rb

```diff
@@ -7,6 +7,30 @@
     Sprig::Configuration.stub(:new).and_return(configuration)
   end
 
+  describe '#adapter' do
+    let(:supported_adapters) do
+      [:active_record, :mongoid]
+    end
+
+    it { expect(described_class).to respond_to(:adapter).with(0).arguments }
+
+    it { expect(supported_adapters).to include described_class.adapter }
+  end
+
+  describe '#adapter=' do
+    let(:value) { :mongo_mapper }
+
+    it { expect(described_class).to respond_to(:adapter=).with(1).argument }
+
+    it 'changes the value' do
+      previous_value = described_class.adapter
```

Comment:
I'd use an around hook for the resetting to previous value to keep the assertion cleaner here http://www.relishapp.com/rspec/rspec-core/v/2-0/docs/hooks/around-hooks